src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,52 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private Pair<K, V>[] storage;
+
+    public StorageImpl() {
+        storage = new Pair[0];
+    }
+
     @Override
     public void put(K key, V value) {
+        int lengthStorage = storage.length;
+        for (int i = 0; i < lengthStorage; i++) {
+            if (storage[i].getKey() == null && key == null) {
```

Comment:
if they are both null then == will return true
```suggestion
            if (storage[i].getKey() == key) {
```